packages/wrangler/src/sites.tsx

```diff
@@ -148,23 +146,55 @@ export async function syncAssets(
     console.log(`reading ${file}...`);
     const content = await readFile(file, "base64");
 
-    const assetKey = await hashAsset(hasher, file, content);
+    const assetKey = hashAsset(hasher, file, content);
     validateAssetKey(assetKey);
 
     // now put each of the files into kv
     if (!keys.has(assetKey)) {
       console.log(`uploading as ${assetKey}...`);
-      upload.push({
+      toUpload.push({
         key: assetKey,
         value: content,
         base64: true,
       });
     } else {
       console.log(`skipping - already uploaded`);
     }
+    // remove the key from the set so we know we've seen it
+    keys.delete(assetKey);
     manifest[path.relative(siteAssets.baseDirectory, file)] = assetKey;
   }
-  await putBulkKeyValue(accountId, namespace, upload, () => {});
+
+  // `keys` now contains the assets that we need to expire
+  const FIVE_MINUTES_LATER = Date.now() + FIVE_MINUTES;
+  let toExpire: KeyValue[] = [];
+  for (const asset of result) {
+    if (keys.has(asset.name)) {
+      console.log(`expiring unused ${asset.name}...`);
+      toExpire.push({
+        key: asset.name,
+        value: "", // we'll fill all the values in one go
+        expiration: asset.expiration || FIVE_MINUTES_LATER,
```

Comment:
Did you consider using `expiration_ttl` instead?

Reply:
I was just lazy to write the math for a POC PR haha

Reply:
it's an asset in one of the kv namespaces created by wrangler, so we can assume the expiration tags aren't that way.

Reply:
and if it is something out of the ordinary, we should probably not overwrite it because it was explicitly set that way

Reply:
In that case we should also check `expiration_ttl` because setting `expiration` will override that.

Reply:
I believe the api normalises the value and only returns `expiration`, even if you set `expiration_ttl` (I'll have to check, but I vaguely remember a conversation about this)

Reply:
OK - that makes sense. I'll try it out tomorrow to confirm.

Reply:
Indeed if you use `ttl` then the backend converts that to an expiration.

Reply:
So if the asset already has an expiration then we should not be putting it back at all, right?
And moreover we don't need to download its value.

Reply:
Good point, yeah we can exclude it from the set if it already has an expiration value on it